### ISSUE TITLE
Create target package for supervised target visualizers

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#334]
For supervised machine learning, there is some independent analysis that must be conducted on the target vector; for example label encoding, "bucketization", etc. 

For visualizers:

- move the class balance here without requiring a model, fit, or score 
- perform detection of sequential or categorical data (and associated colors) 
- plot feature/target correlations in this package

**Comment by bbengfort** [comment]
Can also create a mutual information with target visualizer:

http://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_classif.html#sklearn.feature_selection.mutual_info_classif
